_posts/2020-11-30-flexible-and-safe-tuples-in-scala3.md

```diff
@@ -0,0 +1,387 @@
+---
+layout: blog-detail
+post-type: blog
+by: Vincenzo Bazzucchi, Scala Center
+title: Flexible and safe tuples in Scala 3
+---
+
+# Flexible and safe tuples in Scala 3
+
+Tuples are completely revisited in Scala 3.
+They are more **flexible**, more dynamic and support a **new range of operations**.
+This is enabled by new and powerful features in the Scala 3 compiler.
+
+In this post we will explore the new capabilities of tuples before
+looking under the hood to learn how the improvements in the Scala 3
+compiler enable to implement type safe operations on tuples.
+
+# The basics: what are tuples?
+
+In the Python programming language, tuples are a simple concept:
+they are immutable collections of objects. As such, they are opposed
+to lists, which are mutable.
```

Comment:
I started with Python because it is the only main stream language providing tuples. I could have started with Rust or Haskell but probably they are less well known

Reply:
(It's not so much the choice of Python specifically I'm reacting to, but the choice to lead with a comparison to any other programming language at all. Anyway, no big deal.)